### PR TITLE
feat: create azuredevops_project_pipeline_settings resource

### LINKED ISSUE #410: New Resource: Pipeline general settings
When creating a build definition that need to access sources from a different project we are getting this error.

The current workaround is to change those 3 settings:
image

### PR DESCRIPTION
**Pull request by josh-barker**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

This PR creates a new resource - `azuredevops_project_pipeline_settings` - which addresses #410

Note: linting warns about using `d.GetOkExists`, but I need to use the function to determine if the `bool` property has been specified in the resource.

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: azuredevops/internal/service/core/resource_project_pipeline_settings.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	clients := m.(*client.AggregatedClient)
+
+	projectId := d.Id()
+	getSettings := build.GetBuildGeneralSettingsArgs{
+		Project: converter.String(projectId),
+	}
+
+	buildSettings, err := clients.BuildClient.GetBuildGeneralSettings(ctx, getSettings)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("Error reading project build general settings: %v", err))
```
Comment (xuzhang3) [inline azuredevops/internal/service/core/resource_project_pipeline_settings.go:93]:
```suggestion
if utils.ResponseWasNotFound(err) {
			d.SetId("")
			return nil
		}
		return diag.FromErr(fmt.Errorf("Error reading project build general settings: Proejct: %s, Error: %v ", projectId, err))
```

File: website/docs/r/project_pipeline_settings.html.markdown [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+- `status_badges_are_private` - (Optional) Disable anonymous access to badges.
+
+> **NOTE:**  
+> The settings at the organization will override settings specified on the project.
+> For example, if `enforce_job_scope` is true at the organization, the `azuredevops_project_pipeline_settings` resource cannot set it to false.
+> In this scenario, the plan will always show that the resource is trying to change `enforce_job_scope` from `true` to `false`.
+
+## Attributes Reference
+
+No attributes are exported
```
Comment (xuzhang3) [inline website/docs/r/project_pipeline_settings.html.markdown]: As all configurations are marked with `Computed`, these properties can be exported.

**Comment by xuzhang3** [comment]
@josh-barker A few changes requested, other LGTM.  
For the PAT permissions,  according my test, the PAT need full access permission or you will get 401 unauthorized error.  This is a service bug.
Ref: https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/manage-pats-with-policies-for-administrators?view=azure-devops#restrict-creation-of-full-scoped-pats

**Comment by josh-barker** [comment]
Hey @xuzhang3, I can't see your feedback in the PR. Happy to address the requested changes.

**Review by xuzhang3 — CHANGES_REQUESTED** [review]

**Comment by xuzhang3** [comment]
@josh-barker Sorry, forgot to submit.

**Comment by josh-barker** [comment]
Hey @xuzhang3, I've addressed your feedback.

Cheers

**Comment by xuzhang3** [comment]
LGTM,. shipping it  ⛵

**Pull request merged**